trackintel/model/util.py

```diff
@@ -152,29 +152,103 @@ def _constructor(self):
         """Interface to subtype pandas properly"""
         # we loose access to self in inner function -> write objects to local variables
         super_cons = super()._constructor
-        class_cons = self.__class__
-        check = self._check
 
         def _constructor_with_fallback(*args, **kwargs):
             result = super_cons(*args, **kwargs)
             # cannot validate_geometry as geometry column is maybe not set
-            if isinstance(result, GeoDataFrame) and check(result, validate_geometry=False):
-                return class_cons(result, validate_geometry=False)
+            if isinstance(result, GeoDataFrame) and self._check(result, validate_geometry=False):
+                return self.__class__(result, validate_geometry=False)
             return result
 
         return _constructor_with_fallback
 
 
-class TrackintelBase(object):
-    """Class for supplying basic functionality to all Trackintel classes."""
+def _wrapped_gdf_method_fallback(func):
+    """Decorator function that downcast types to trackintel class if is (Geo)DataFrame and has the required columns."""
+
+    @wraps(func)  # copy all metadata
+    def wrapper(self, *args, **kwargs):
+        result = func(self, *args, **kwargs)
+        if isinstance(result, pd.DataFrame) and self._check(result, validate_geometry=False):
+            if isinstance(result, GeoDataFrame):
+                result.__class__ = self.__class__
+            else:
+                result.__class__ = self.fallback_class
+        return result
+
+    return wrapper
 
-    # so far we don't have a lot of methods here
-    # but a lot of IO code can be moved here.
+
+# short and memorizable name
```

Comment:
Is this still a todo?

Reply:
Yes :). I'll open an issue to remind us to fix this.